### ISSUE TITLE
"Clear All Categories" Button

### ISSUE THREAD

**Issue by RajChaudhry** [codeforboston/communityconnect#339]
Currently, we have to clear each category individually when searching within the admin page. Having a button which "unchecks" all the categories at the same time would make use much easier.

**Comment by FMA126** [comment]
Working on this

**Comment by RajChaudhry** [comment]
Hey! I was looking at the code, and I found a function in the Header.js written which has this functionality. To trigger it, make a button in React, where the onClick event is `modal.toggle`. It opens a modal confirming that you want to clear all resources, and upon clicking it, it'll redirect you to /admin . Take a look there for help structuring the code!

**Comment by FMA126** [comment]
If I select a few resources and then open the modal and press continue, I'm getting an error 404.  Does the continue button just redirect to /admin and close the modal? 

If the button I'm working on to 'uncheck' all the categories just refreshes the page, how would I keep the previous state of the saved resources the same?

Thanks for helping, I'm learning about react this week in school.

**Comment by RajChaudhry** [comment]
Wait I'm sorry, I was mixing this issue up with another issue I wanted to open up later. Sorry about the confusion, ignore what I said. This is what happens when I get distracted during class.

**Comment by FMA126** [comment]
I think I found a bug? In the header.js file
```
  confirmationModalToggle = () => {
    window.location.href = "/admin";
    this.modalToggle();
  };
```
I think `window.location.href` has to be set equal to `'/revere/admin'`

If I do this with the clear categories button it works but we lose the state of the saved resources

**Comment by FMA126** [comment]
I'm putting in the pr now

**Comment by RajChaudhry** [comment]
> I think I found a bug? In the header.js file
> 
> ```
>   confirmationModalToggle = () => {
>     window.location.href = "/admin";
>     this.modalToggle();
>   };
> ```
> 
> I think `window.location.href` has to be set equal to `'/revere/admin'`
> 
> If I do this with the clear categories button it works but we lose the state of the saved resources

This is a bug that we know, however I neglected to create an issue for it. Thank you for reminding me!

**Comment by FMA126** [comment]
Hey everybody, I just updated this pr. Not sure if anyone has looked at this yet but I solved a few issues of the previous pr but the issue hasn't been solved yet.  I'll keep researching how to do this.

**Comment by FMA126** [comment]
Just submitted a pull request.  I ended up refactoring the whole component and added some life cycle methods.  The previous way we were filtering the resources was really confusing and it was changing the state incorrectly by mutating state.  My refactor uses setState() to change the state correctly.  I also changed the styling and replaced the check boxes with list items.  The check boxes are a massive effort to get to work and clear etc.  Hopefully everyone likes the new component !
